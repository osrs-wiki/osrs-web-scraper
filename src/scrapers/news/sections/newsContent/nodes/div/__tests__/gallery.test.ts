import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import fs from "fs";
import parse from "node-html-parser";

import galleryParser from "../gallery";

// Mock file system operations
jest.mock("fs");

describe("gallery node", () => {
  test("Gallery should parse and render", () => {
    const existsSyncSpy = jest
      .spyOn(fs, "existsSync")
      .mockImplementationOnce(() => false);
    const mkdirSyncSpy = jest
      .spyOn(fs, "mkdirSync")
      .mockImplementationOnce(() => "");

    const root = parse(`
      <div class="row">
        <img src="https://example.com/image1.png" />
        <img src="https://example.com/image2.jpg" />
      </div>
    `);

    const builder = new MediaWikiBuilder();
    builder.addContents([galleryParser(root, { title: "test-title" })].flat());
    expect(builder.build()).toMatchSnapshot();

    expect(existsSyncSpy).toHaveBeenCalledWith(`./out/news/test-title`);
    expect(mkdirSyncSpy).toHaveBeenCalledWith(`./out/news/test-title`, {
      recursive: true,
    });
  });
});
