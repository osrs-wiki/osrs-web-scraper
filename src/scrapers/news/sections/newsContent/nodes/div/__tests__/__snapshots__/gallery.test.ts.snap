// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gallery node Gallery should parse and render 1`] = `
"<gallery mode=\\"packed\\", heights=\\"180\\", style=\\"text-align:center\\">
test-title (1).png
test-title (2).jpg
</gallery>
"
`;
